src/hotspot/share/gc/shenandoah/shenandoahFullGC.cpp

```diff
@@ -1334,7 +1343,8 @@ void ShenandoahFullGC::compact_humongous_objects() {
         ShenandoahRegionAffiliation original_affiliation = r->affiliation();
         for (size_t c = old_start; c <= old_end; c++) {
           ShenandoahHeapRegion* r = heap->get_region(c);
-          r->make_regular_bypass();
+          // Consider regions emptied by humongous evacuation to be young
+          r->make_regular_bypass(true);
```

Comment:
If we remove the side-effect of `make_regular_bypass`, we would have something like:
```java
if (r->is_humongous_start() || r->is_humongous_cont()) {
  r->set_affiliation(YOUNG);
}
```

Reply:
I've implemented and tested a change that removes the change-affiliation side effect of make_regular_bypass.